### PR TITLE
Added JUnit tests for Metatype package - part1

### PR DESCRIPTION
**Pull request by code-incubus**
Signed-off-by: code-incubus nemanja.milutinovic@comtrade.com

Added JUnit tests for classes in a metatype package:

- EmptyTocdTest.java
- TadImplTest.java
- TattributeImplTest.java
- TdesignateImplTest.java
- TiconImplTest.java
- ToptionImplTest.java

Related Issue
None

Description of the solution adopted
/
Screenshots
/

Any side note on the changes made
/

### CODE REVIEW

**Review by LeoNerdoG — APPROVED** [review]

**Comment by LeoNerdoG** [comment]
Hey @Coduz,
I have reviewd this branch and it looks OK. Can you plese merge it into develop branch? All the tests have successfully passed.

File: commons/src/test/java/org/eclipse/kapua/commons/configuration/metatype/EmptyTocdTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    List<Object> objList = Arrays.asList(
+            new EmptyTocd("1", "name"),
+            new EmptyTocd("", ""),
+            new EmptyTocd("2", ""),
+            new EmptyTocd("", "name2"));
+
+    @Test
+    public void emptyTocdTest() {
+        EmptyTocd emptyTocd = new EmptyTocd("49", "name");
+        assertEquals("49", emptyTocd.getId());
```
Comment (Coduz) [inline commons/src/test/java/org/eclipse/kapua/commons/configuration/metatype/EmptyTocdTest.java]: While using `assertEquals`, `assertNotEquals` (and so on) they al support a message linked to the `assert*` as a first argument. It is a good practice to add a message that shows which specific assert has failed.

So in this case:
`assertEquals("tocd.id", "49", emptyTocd.getId());`
would make any failure linked to a meaningful message which can avoid to have to look at the code to check what has failed.

**Review by Coduz — CHANGES_REQUESTED** [review]

**Comment by code-incubus** [comment]
Hi @Coduz,
I made all the changes that you mentioned above. If something is not ok please let me know.
I'll fix it up quickly.

**Comment by LeoNerdoG** [comment]
Hey @Coduz,
now this PR looks ok, I think this can be merged.

**Review by Coduz — APPROVED** [review]

**Pull request merged**